syncthing-bep/src/main/kotlin/net/syncthing/java/bep/BlockPuller.kt

```diff
@@ -51,10 +53,12 @@ object BlockPuller {
         val (newFileInfo, fileBlocks) = indexHandler.getFileInfoAndBlocksByPath(fileInfo.folder, fileInfo.path) ?: throw FileNotFoundException()
 
         if (fileBlocks.hash != fileInfo.hash) {
-            throw IllegalStateException("the current file entry hash does not match the hash of the provided one")
+            val exception = IllegalStateException("The current file entry hash does not match the hash of the provided one.")
+            LOGGER.atInfo().withThrowable(exception).log("Expected hash: ( {} ), Provided hash: ( {} ).",fileInfo.hash,fileBlocks.hash)
+            throw exception
         }
 
-        logger.info("pulling file = {}", fileBlocks)
+        LOGGER.atInfo().log("Pulling File: {}, File Blocks: {}.", fileInfo.path, fileBlocks)
```

Comment:
Look at the `toString()` method of `fileBlocks`. It already includes a `path`. Is that the same path as `fileInfo.path`?

Reply:
I think so. This has been an adventure into Kotlin for me... but I think, from tracing this back thru the `getFileInfoAndBlocksByPath` function and the Class definitions of `FileBlocks.kt` & `FileInfo.kt`, that there is some duplicate information being passed in the `Pair.of(fileInfo, fileBlocks)`.

Reply:
Is this correctly interpreting the Kotlin function?
`fun getFileInfoAndBlocksByPath(folder: String, path: String): Pair<FileInfo, FileBlocks>? {`
means:
Received arguments: `folder`, `path` (both strings)
`:` --> means Returns
Returned arguments: `<FileInfo, FileBlocks>` (as pair, defined by their respective class definitions)
`?` --> means can return `null`

Reply:
yes, that's correct. nicely done